### PR TITLE
Config API

### PR DESCRIPTION
**Pull request by snehal0904**

### CODE REVIEW

**Comment by coolbung** [comment]
Let us not expose the internal / increment id in the API. Otherwise, when updating, it'll always be needed to read the config, get the IDs and then update based on IDs. Let's avoid this and manage it in the create/update API. 

Let us work only with the `module` and `key`. The combination of module + key + context id will be always unique. So think if the below endpoints only

GET /config/:module/all
POST /config/:module
This API will receive an array of key, value, context, context_id and create/update the configs. If a config exists it'll be updated. Otherwise create it.

**Comment by snehal0904** [comment]
> 

Added the changes in the post API, it will check if the record exists update else create.
Should I remove, update by ID API?

**Comment by coolbung** [comment]
Yes. Let's have only the 2 APIs mentioned in my previous comment

**Comment by snehal0904** [comment]
Changes done

File: src/adapters/default/config.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        filters: {
+          contextId: {
+            eq: `${schoolId}`,
+          },
+        },
+      };
+
+      let final = {
+        method: "post",
+        url: `${this.url}/search`,
```
Comment (coolbung) [inline src/adapters/default/config.adapter.ts:139]: If the school specific config is not available it should use the global one. This does not seem to be implemented ?

File: src/configs/config.controller.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  @Post(":module")
+  @ApiBasicAuth("access-token")
+  @ApiCreatedResponse({ description: "Config has been created successfully." })
+  @ApiBody({ type: ConfigDto })
+  @ApiForbiddenResponse({ description: "Forbidden" })
+  @UseInterceptors(ClassSerializerInterceptor)
+  public async createConfig(
+    @Req() request: Request,
+    @Body() configDto: ConfigDto
```
Comment (coolbung) [inline src/configs/config.controller.ts:55]: This seems to be saving one config at a time. Is there no way to make it save all the configs in a single API call ?

File: src/adapters/default/config.adapter.ts (hunk 1 above)
Comment (snehal0904) [reply src/adapters/default/config.adapter.ts:136]: Sent global config, if School specific not found

File: src/configs/config.controller.ts (hunk 2 above)
Comment (snehal0904) [reply src/configs/config.controller.ts:55]: Currently, it is saving one config at a time. If we want to save multiple rows at a time, we will have to take request data and traverse a loop and save it.

File: src/configs/interfaces/config.interface.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export interface ConfigInterface {
```
Comment (coolbung) [inline src/configs/interfaces/config.interface.ts:1]: Are we using interfaces @ManojNathIC  ?

File: src/configs/dto/config-response.dto.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export interface ConfigResponseDto {
```
Comment (coolbung) [inline src/configs/dto/config-response.dto.ts:1]: The other DTOs are classes, this should also be a class, not interface right ?

File: src/configs/interfaces/config.interface.ts (hunk 3 above)
Comment (ManojNathIC) [reply src/configs/interfaces/config.interface.ts:1]: No we are using DTO only

File: src/adapters/default/config.adapter.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+      var updateConfig = {
+      method: "put",
+      url: `${this.url}/${configId}`,
+      headers: {
+        Authorization: request.headers.authorization,
+      },
+      data: udateData,
+      };
+      const response = await axios(updateConfig);
+      console.log("123");
```
Comment (coolbung) [inline src/adapters/default/config.adapter.ts:62]: Remove these logging statements please. Instead you can print a proper log message that can be used for debugging later.

**Pull request merged**